### ISSUE TITLE
settings.py is too confusing

### ISSUE THREAD

**Issue by ronaldtse** [ietf-ribose/bibxml-service#42]
The config file should be simple to use. Right now it mixes many different concepts and makes it hard to understand/use/config.

For example, I can't tell the difference between these:

```
DATASET_SOURCE_OVERRIDES
AUTHORITATIVE_DATASETS
EXTERNAL_DATASETS
KNOWN_DATASETS
LEGACY_DATASETS
```

If I just want to do #40 or #41 , what do I do? The file does not answer this.

**Comment by strogonoff** [comment]
Settings are minimally documented for a maintainer already aware how things work generally. (Note that configuration is not intended to be edited by non-maintainers.)

But yes, documentation is very insufficient, and this certainly needs to be addressed (#44, #45).

In meantime, a quick additional summary if that helps:

* DATASET_SOURCE_OVERRIDES — configures dataset source, only if it’s different than [default](https://github.com/ietf-ribose/bibxml-service/issues/41#issuecomment-998347891).
* AUTHORITATIVE_DATASETS — which dataset are authoritative (by the criteria you have provided, authoritative datasets are considered i-ds, rfcs and rfcsubseries)
* EXTERNAL_DATASETS — which of the known datasets are external sources (not indexable, like DOI and later Datatracker)
* KNOWN_DATASETS — which dataset IDs do we have at all
* LEGACY_DATASETS — dataset IDs for xml2rfc compatibility APIs. Its documentation spans almost 100 lines, so I doubt it’s hard to tell how _this_ one is used.

**Comment by strogonoff** [comment]
The latest update changes settings:

- now we have ``RELATON_DATASETS``, ``EXTERNAL_DATASETS`` and ``AUTHORITATIVE_DATASETS``
- documentation now references the settings
